app/models/replication_transfer.rb

```diff
@@ -21,10 +21,10 @@ def self.find_fields
 
 
 
-  def cancel!(reason)
+  def cancel!(reason, detail)
     unless cancelled
       transaction do
-        update!(cancelled: true, cancel_reason: reason)
+        update!(cancelled: true, cancel_reason: reason, cancel_reason_detail: detail)
         bag_man_request&.cancel!(reason)
       end
     end
```

Comment:
The control flow, transactionality or atomicity, of the `cancel!` operations between the `ReplicationTransfer` and the `BagManRequest` are complex because they each call the other and, without the `unless cancelled` wrapper, might incur an infinite loop.  I wonder whether the calls from one model to the other should be extracted out of the `transaction` block, to allow an atomic update of one model and then another model?  Or, perhaps this coupling between these models should be extracted from the model level up to the controller level?